java/src/main/java/ai/onnxruntime/OrtUtil.java

```diff
@@ -41,9 +41,9 @@ public static int[] transformShape(long[] shape) {
     int[] newShape = new int[shape.length];
     for (int i = 0; i < shape.length; i++) {
       long curDim = shape[i];
-      if (curDim < 1 || curDim > Integer.MAX_VALUE) {
+      if (curDim < 0 || curDim > Integer.MAX_VALUE) {
```

Comment:
>urDim < 0 [](http://example.com/codeflow?start=11&length=9)

dimension that has a value of -1 indicates that this is a symbolic dimension that has a symbolic name in that position, but otherwise it is a valid dimension.

I think it would be beneficial to run onnx models with Java API and see any adjustments necessary.

Reply:
Yeah, we have tests for symbolic dimension behaviour - https://github.com/microsoft/onnxruntime/blob/main/java/src/test/java/ai/onnxruntime/InferenceTest.java#L576. Everything I've tested works, but the initial version of this code I wrote many years ago didn't cope with zero length tensors because I couldn't figure out why they would occur. This change is to rewrite some of that validation and allow them through.